gobotics-frontend/src/App.tsx

```diff
@@ -32,23 +32,30 @@ type AppState = {
 
 class App extends Component<AppProps, AppState>{
     constructor(props:AppProps) {
-        super(props)
+        super(props);
+        fetch('/client/devices').
```

Comment:
doesn't matter now since you only have one call, but this should eventually be factored out into a different function/file.
You should have a file for API calls (or you could take this as an opportunity to explore redux).
App.tsx shouldn't have to know anything about executing the request (URL), it should just act on the promise object.